doc/generating-code.mld

```diff
@@ -0,0 +1,357 @@
+{%html: <div style="display: flex; justify-content:space-between"><div>%}{{!"writing-ppxs"}< Writing PPXs}{%html: </div><div>%}{{!"matching-code"}Destructing AST nodes >}{%html: </div></div>%}
+
+{0 Generating AST Nodes}
+
+The rewriter's core is a function that outputs code in the form of an AST.
+However, there are some issues with generating AST values when using the
+constructors directly:
+
+- The type is {{!Ppxlib.Parsetree}pretty verbose}, with many fields
+  rarely used.
+- The AST type might change at a version bump. In this case, the types used in
+the PPX would become incompatible with the types of the new OCaml version.
+
+The second point is important: since [ppxlib] {{!page-driver.compat_mult_ver}translates} the AST to
+the newest OCaml AST available before rewriting, your PPX would not
+only become incompatible with the new OCaml version, but also with all [ppxlib]
+versions released after the new AST type is introduced.
+
+For this reason, [ppxlib] provides abstractions over the OCaml AST, with a focus
+on usability and stability.
+
+{1 The Different Options}
+
+The two main options are:
+
+- {{!Ppxlib.Ast_builder}[Ast_builder]},
+- {!Ppxlib_metaquot}.
+
+{{!Ppxlib.Ast_builder}[Ast_builder]} provides an API to generate AST nodes for the latest OCaml
+version in a backward-compatible way. {!Ppxlib_metaquot} is different: it is a PPX that
+lets you generate OCaml AST nodes by writing OCaml code, using quoting and
+splicing.
+
+Using {!Ppxlib_metaquot} requires less knowledge of the OCaml AST than
+{{!Ppxlib.Ast_builder}[Ast_builder]} as it only uses natural OCaml syntax;
+however, it is less general because it only allows users to generate few different nodes such as
+structure items, expressions, patterns, etc., but it is not possible to generate a
+value of type {{!Ppxlib.Parsetree.row_field_desc}[row_field_desc]}!
+
+Note: The OCaml compiler API also provides a module, called {{!Ppxlib.Ast_helper}[Ast_helper]}, to
```

Comment:
I would either remove this or explain that it's preferred to use Ast_builder when you need fine granted Nodes

Reply:
I am not sure about the status between Ast_helper and Ast_builder, which is why I was vague but I still wanted to mention it since it is in the API. As suggested, I recommended Ast_builder.